packages/@react-spectrum/table/src/Table.tsx

```diff
@@ -441,11 +444,13 @@ function TableRow({item, children, ...otherProps}) {
   let ref = useRef();
   let state = useTableContext();
   let isSelected = state.selectionManager.isSelected(item.key);
+  let isDisabled = state.disabledKeys.has(item.key);
```

Comment:
There is a useHover on line 463 which you can pass this `isDisabled` to disable the hover styles on disabled rows